src/bosh-director/lib/bosh/director/vm_creator.rb

```diff
@@ -86,6 +86,9 @@ def get_agenda_for_instance_plan(instance_plan, disks, tags, ip_provider, total,
         @dns_encoder,
         @link_provider_intents,
       )
+      if Config.enable_short_lived_nats_credentials
+        agenda.steps << DeploymentPlan::Steps::PermanentNatsCredentialsStep.new
```

Comment:
This feels counter-intuitive, to add a permanent nats credential only when using short-lived creds.

Reply:
Yes, reads much more intuitively 👍